DNN Platform/Website/Default.aspx.cs

```diff
@@ -632,8 +632,29 @@ private void InitializePage()
             // register the custom stylesheet of current page
             if (this.PortalSettings.ActiveTab.TabSettings.ContainsKey("CustomStylesheet") && !string.IsNullOrEmpty(this.PortalSettings.ActiveTab.TabSettings["CustomStylesheet"].ToString()))
             {
-                var customStylesheet = Path.Combine(this.PortalSettings.HomeDirectory, this.PortalSettings.ActiveTab.TabSettings["CustomStylesheet"].ToString());
-                ClientResourceManager.RegisterStyleSheet(this, customStylesheet);
+                var customStylesheet = this.PortalSettings.ActiveTab.TabSettings["CustomStylesheet"].ToString();
+                var cssIncluded = false;
+                
+                // Try and go through the FolderProvider first, if it gives back an external url, include it
+                // as a <link> in the Page Header
+                IFileInfo fi = GetPageStylesheetFileInfo();
+                if (fi != null)
+                {
+                    string url = FileManager.Instance.GetUrl(fi);
+                    if (url.StartsWith("http://", StringComparison.InvariantCultureIgnoreCase) || url.StartsWith("https://", StringComparison.InvariantCultureIgnoreCase))
+                    {
+                        string cssLink = string.Format("<link href=\"{0}\" rel=\"stylesheet\" type=\"text/css\" />", url);
+                        this.Page.Header.Controls.Add(new LiteralControl(cssLink));
+                        cssIncluded = true;
+                    }
+                }
+
+                // Include it as DNN always has, as a relative path through ClientResourceManager
+                if (!cssIncluded)
+                {
+                    var customStylesheetPath = Path.Combine(this.PortalSettings.HomeDirectory, customStylesheet);
+                    ClientResourceManager.RegisterStyleSheet(this, customStylesheetPath);
+                }

```

Comment:
`RegisterStyleSheet` can handle external URLs, so I don't think we need different code paths here
```suggestion
                var stylesheetFile = GetPageStylesheetFileInfo();
                if (stylesheetFile != null)
                {
                    ClientResourceManager.RegisterStyleSheet(this, FileManager.Instance.GetUrl(stylesheetFile));
                }
```

Reply:
Thanks Brian! I didn't know RegisterStyleSheet could handle absolute urls.  My only concern is that while any CSS that is referenced "Should" be in the dnn file system.  I image there could be a case where it's not in the DNN filesystem especially since the css field can contain anything the user enters and the DNN filesystem isn't synced.  However if you feel it's best to make the assumption that the css file is in the DNN filesystem, I'd be happy to make the change.  Just want to make sure.

Reply:
Testing out ClientResourceManager.RegisterStyleSheet(), it doesn't seem to take absolute paths.

Message:'https:/folderprovider.blob.core.windows.net/dnnfork/Cloud%202/test.css' is not a valid virtual path.

StackTrace:

InnerMessage:'https:/folderprovider.blob.core.windows.net/dnnfork/Cloud%202/test.css' is not a valid virtual path.

InnerStackTrace:

   at System.Web.Util.UrlPath.CheckValidVirtualPath(String path)
   at System.Web.Util.UrlPath.Combine(String appPath, String basepath, String relative)
   at System.Web.VirtualPath.Combine(VirtualPath relativePath)
   at System.Web.HttpRequest.MapPath(VirtualPath virtualPath, VirtualPath baseVirtualDir, Boolean allowCrossAppMapping)
   at System.Web.HttpServerUtility.MapPath(String path)
   at DotNetNuke.Web.Client.ClientResourceManagement.ClientResourceManager.RegisterStyleSheet(Page page, String filePath, Int32 priority, String provider, String name, String version) in C:\dnnsites\dnnfork\DNN Platform\DotNetNuke.Web.Client\ClientResourceManager.cs:line 303
   at DotNetNuke.Web.Client.ClientResourceManagement.ClientResourceManager.RegisterStyleSheet(Page page, String filePath, Int32 priority, String provider) in C:\dnnsites\dnnfork\DNN Platform\DotNetNuke.Web.Client\ClientResourceManager.cs:line 282
   at DotNetNuke.Web.Client.ClientResourceManagement.ClientResourceManager.RegisterStyleSheet(Page page, String filePath) in C:\dnnsites\dnnfork\DNN Platform\DotNetNuke.Web.Client\ClientResourceManager.cs:line 248
   at DotNetNuke.Framework.DefaultPage.InitializePage() in C:\dnnsites\dnnfork\DNN Platform\Website\Default.aspx.cs:line 644
   at DotNetNuke.Framework.DefaultPage.OnInit(EventArgs e) in C:\dnnsites\dnnfork\DNN Platform\Website\Default.aspx.cs:line 187
   at System.Web.UI.Control.<InitRecursiveAsync>d__225.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Web.Util.WithinCancellableCallbackTaskAwaitable.WithinCancellableCallbackTaskAwaiter.GetResult()
   at System.Web.UI.Page.<ProcessRequestMainAsync>d__523.MoveNext()

Reply:
You're right about the file potentially being external, I've added an `else` block to my suggestion.

The error you've hit is line 303 here, which is inside the `if (filePath.Contains(".css?"))` block.  I think we need to add some better error handling in there, because the addition of a query string to the URL shouldn't mean that it has to be a valid virtual path.
https://github.com/dnnsoftware/Dnn.Platform/blob/edb0facc4253c93a4ef3bdd974f599506de56258/DNN%20Platform/DotNetNuke.Web.Client/ClientResourceManager.cs#L296-L323

We could adjust that code like this:
```
if (!Uri.TryCreate(filePath, UriKind.Absolute, out _) && filePath.Contains(".css?"))
```

Reply:
Thanks Brian, I have refactored/applied the changes.  Let me know... thanks